diff --git a/bridges/source/cpp_uno/gcc3_linux_aarch64/cpp2uno.cxx b/bridges/source/cpp_uno/gcc3_linux_aarch64/cpp2uno.cxx
index e4f7029e6..48e768c63 100644
--- a/bridges/source/cpp_uno/gcc3_linux_aarch64/cpp2uno.cxx
+++ b/bridges/source/cpp_uno/gcc3_linux_aarch64/cpp2uno.cxx
@@ -583,7 +583,8 @@ unsigned char * bridges::cpp_uno::shared::VtableFactory::addLocalFunctions(
 void bridges::cpp_uno::shared::VtableFactory::flushCode(
     unsigned char const * begin, unsigned char const * end)
 {
-#if !defined ANDROID && !defined MACOSX
+// A static executable does not export __clear_cache for dlsym to find.
+#if !defined ANDROID && !defined MACOSX && !defined DISABLE_DYNLOADING
     static void (*clear_cache)(unsigned char const *, unsigned char const *)
         = reinterpret_cast<void (*)(unsigned char const *, unsigned char const *)>
             (dlsym(RTLD_DEFAULT, "__clear_cache"));
